crates/evm-tracing-client/src/serialization.rs

```diff
@@ -50,15 +51,11 @@ where
 }
 
 /// Serializes opcode.
-pub fn opcode_serialize<S>(opcode: &[u8], serializer: S) -> Result<S::Ok, S::Error>
+pub fn opcode_serialize<S>(opcode: &MarshalledOpcode, serializer: S) -> Result<S::Ok, S::Error>
 where
     S: Serializer,
 {
-    let d = std::str::from_utf8(opcode)
-        .map_err(|_| S::Error::custom("Opcode serialize error."))?
-        .to_uppercase();
-
-    serializer.serialize_str(&d)
+    serializer.serialize_str(&opcode.to_string())
```

Comment:
Code before didn't malloc (used `str`); now it does (goes through `String`); two questions here @dmitrylavrenov.

1. Do you understand the difference?
2. Why change it this way?

Reply:
it comes from `Display` implementation that uses malloc